### DISCUSSION TITLE
my backups have been failing since I moved them off the local drive

### DISCUSSION THREAD

**Discussion by y8s** [nextcloud/all-in-one#5692] [Questions] [answered]
I was running out of space on my VM and realized that my AIO backup dir was set to a dir in my /mnt folder that wasn't actually mounted -- making all my backups on the same filesystem. Not what i wanted.

I reset the backup location in the AIO interface to something ACTUALLY mounted via NFS and tried to make a backup. I got a bunch of stale file handle errors. I tried starting over, deleting the borg cache, but now it fails with other errors when creating the repo.

I have also tried using a SMB share and borgbase but each time it works on the backup for a few hours then fails at the end and deletes the backup. On borgbase, it was one stale file handle that broke it.

I really want to use NFS (example `/mnt/backup/nc-aio` as my dir). I set that up and here's my log:

[borglog.txt](https://github.com/user-attachments/files/17980952/borglog.txt)

Help is appreciated!

**Comment by y8s** [comment]
update: 
I nuked the existing backup location and config and started fresh after realizing I had a permission error. The  repo is created now and starts but after a while every line looks like this:

`2024-12-02T18:05:07.729529751Z /nextcloud_aio_volumes/nextcloud_aio_nextcloud_data/user/files/InstantUpload/DCIM/2021/12/Camera/PXL_20211215_024308327.jpg: read: [Errno 116] Stale file handle`
So I know it will fail but I still have to wait a few hours for it to finish and then tell me it didn't work. Would be nice to have it stop when this first happens instead of at the end of thousands of lines of errors.

Is this an NFS limitation?
Would I see it also with SMB?
What about remote (borgbase)?

**Comment by szaimen** [comment]
Hi, I guess it would make sense to open an issue about this at https://github.com/borgbackup/borg and post the exact error message there.

**Comment by y8s** [comment]
I have other borg repos on the same system that work fine over NFS. I also don't see other people with this issue. 

Does aio need special permission or uid gid set in the master container? I have had to do that for other docker stuff where data resides on an NFS share. weirdly aio doesn't care that the main nextcloud file storage is on the same server.

**Comment by y8s** [comment, marked as answer]
just successfully backed up to NFS.
I looked around the Internet and found similar issues with the stale file handles and some suggestions were to increase the  file lease to 5 minutes. 

I went in to nfs.conf and raised the lease and grace to 5 mins, re-exported the shares and started a backup. 

it worked thankfully.

still unclear why a local backup fails though.
